### DISCUSSION TITLE
How to migrate stored data from v0.2 to v0.3?

### DISCUSSION THREAD

**Discussion by viiiprock** [facebook/lexical#2414] [Q&A] [answered]
At v0.2, I stored my data as "{_nodeMap: [....]}" with `editorState.toJSON()` , now I'm struggling with how to migrate to v0.3
when use `editor.parseEditorState` it throws an error `Cannot read properties of undefined (reading 'type').`
Anyone can help me out? Many thanks.

**Comment by trueadm** [comment, marked as answer]
You can parse the old JSON format using this function: https://github.com/facebook/lexical/blob/main/packages/lexical-utils/src/index.ts#L252

Then you can create a new JSON format using the updated format in v0.3,

**Reply by viiiprock** [reply]
Thanks @trueadm 
I think I can make it work with `unstable_convertLegacyJSONEditorState` or `unstable_parseEditorState`, btw, some type may not existed, for example in my case `image`, I guess I have to copy and make it fit for the custom plugin
You can see my attempt here 
https://codesandbox.io/s/lexical-v3-uxdl39?file=/src/RichText.jsx
with the `dataV2` example
